test/integration/api/forms/forms.js

```diff
@@ -324,6 +297,156 @@ describe('api: /projects/:id/forms (create, read, update)', () => {
+        const asAlice = await service.login('alice');
+        global.enketo.state = 'error';
+        const { body } = await asAlice.post('/v1/projects/1/forms')
+          .set('Content-Type', 'application/xml')
+          .send(testData.forms.simple2)
+          .expect(200);
+        should.not.exist(body.enketoId);
+        should.not.exist(body.enketoOnceId);
+      }));
+
+      it('should wait for Enketo only briefly @slow', testService(async (service) => {
+        const asAlice = await service.login('alice');
+        global.enketo.wait = (done) => { setTimeout(done, 600); };
+        const { body } = await asAlice.post('/v1/projects/1/forms')
+          .set('Content-Type', 'application/xml')
+          .send(testData.forms.simple2)
+          .expect(200);
+        should.not.exist(body.enketoId);
+        should.not.exist(body.enketoOnceId);
+      }));
+
+      it('should request an enketoId from worker if request from endpoint fails', testService(async (service, container) => {
+        const asAlice = await service.login('alice');
+
+        // First request to Enketo, from the endpoint
+        global.enketo.state = 'error';
+        await asAlice.post('/v1/projects/1/forms')
+          .set('Content-Type', 'application/xml')
+          .send(testData.forms.simple2)
+          .expect(200);
+
+        // Second request, from the worker
+        global.enketo.callCount.should.equal(1);
+        await exhaust(container);
+        global.enketo.callCount.should.equal(2);
+        const { body } = await asAlice.get('/v1/projects/1/forms/simple2')
+          .expect(200);
+        global.enketo.createData.should.eql({
+          openRosaUrl: `${container.env.domain}/v1/test/${body.draftToken}/projects/1/forms/simple2/draft`,
+          xmlFormId: 'simple2',
+          token: undefined
+        });
+        body.enketoId.should.equal('::abcdefgh');
+        should.not.exist(body.enketoOnceId);
+      }));
+
+      it('should not request an enketoId from worker if request from endpoint succeeds', testService(async (service, container) => {
+        const asAlice = await service.login('alice');
+        await asAlice.post('/v1/projects/1/forms')
+          .set('Content-Type', 'application/xml')
+          .send(testData.forms.simple2)
+          .expect(200);
+        global.enketo.callCount.should.equal(1);
+        await exhaust(container);
+        global.enketo.callCount.should.equal(1);
+      }));
+    });
+
+    describe('Enketo IDs for published form', () => {
+      it('should request Enketo IDs', testService(async (service, { env }) => {
+        const asAlice = await service.login('alice');
+        const { body } = await asAlice.post('/v1/projects/1/forms?publish=true')
+          .set('Content-Type', 'application/xml')
+          .send(testData.forms.simple2)
+          .expect(200);
+        global.enketo.callCount.should.equal(1);
+        without(['token'], global.enketo.createData).should.eql({
```

Comment:
I don't think we need to test the `token` argument at this time. Unlike in the draft case above, `token` won't be `undefined`. However, I'm not sure that the token is even needed for this flow to function: see #997. We don't test the token in existing tests.